### DISCUSSION TITLE
Questions about Pinta

### DISCUSSION THREAD

**Discussion by jokoon** [PintaProject/Pinta#997] [General]
Hello!

First thanks for Pinta, I love paint.net and foobar2000 and paintnet are the 2 software that prevent me to switching to linux full time.

I remember trying it years ago, it crashed regularly, I guess those are fixed.

Happy to see that there's outline text by default, although could I see the code involved with the outline? On paintnet, there is the outline plugin by pyrochild, and it allows the user to pick a width and softness.

![image](https://github.com/user-attachments/assets/a0c7b1c0-ccb3-47aa-984b-37db25ec7be4)

This allows to make better text outline effects.

Also, there are plugins to create gradual transparency effects when needed:

![image](https://github.com/user-attachments/assets/3e294376-eafb-446d-8aef-8715ce903a87)

![image](https://github.com/user-attachments/assets/62f30a5d-7f6f-4082-b1ca-7819cf3440f7)

Thanks for developing Pinta, I guess old crashes are fixed.

**Comment by cameronwhite** [comment]
The text outline code is around here - most of the work happens in the Cairo and Pango libraries though
https://github.com/PintaProject/Pinta/blob/f2eb975c3c2ac997e2d90c447199f90a952dad0c/Pinta.Tools/Tools/TextTool.cs#L1031

You might also be interested in the new Outline Object and Feather Object effects which were very recently added in #971 and #953, which sound somewhat similar to the plugins you mentioned
